src/lib/mdns/Resolver_ImplMinimalMdns.cpp

```diff
@@ -256,8 +421,84 @@ CHIP_ERROR MinMdnsResolver::SetResolverDelegate(ResolverDelegate * delegate)
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR MinMdnsResolver::SendQuery(mdns::Minimal::FullQName qname, mdns::Minimal::QType type)
+{
+
+    System::PacketBufferHandle buffer = System::PacketBufferHandle::New(kMdnsMaxPacketSize);
+    ReturnErrorCodeIf(buffer.IsNull(), CHIP_ERROR_NO_MEMORY);
+
+    QueryBuilder builder(std::move(buffer));
+    builder.Header().SetMessageId(0);
+
+    mdns::Minimal::Query query(qname);
+    query.SetType(type).SetClass(mdns::Minimal::QClass::IN);
+    // TODO(cecille): Not sure why unicast response isn't working - fix.
+    query.SetAnswerViaUnicast(false);
```

Comment:
Against which implementation of mDNS did you observe unicast responses not to work?

Reply:
minimal on m5.

Reply:
Would it be worth keeping SetAnswerViaUnicast to true, but making it do NO-op on minimal DNS client/responder so that when it does work, it will just be an optimal answering over unicast, as opposed to always being multicast?

Reply:
This implementation is a minimal mdns implementation only - other implementations are free to do as they wish. Do you mean just having the minimal mdns advertiser ignore the answer via unicast flag? That seems potentially more dangerous than just opening an issue to look into why this isn't working and changing what we ask for when it's working. Right now, the minimal mdns advertiser indicates it's sending a unicast response when I ask for one, it just doesn't get picked on the resolver side. So it's not clear where the problem actually is.